.github/workflows/check-links-book.yaml

```diff
@@ -0,0 +1,53 @@
+# We also have a check-links-markdown job, however it will not spot mdbook
+# mistakes such as forgetting to list an .md file in SUMMARY.md.
+# Running a link checker on the generated HTML is more reliable.
+
+name: check-links-book
+
+on:
+  pull_request:
+    paths:
+      - "web/book/**"
+      - ".github/workflows/check-links-book.yaml"
+  workflow_call:
+
+env:
+  RUSTFLAGS: "-C debuginfo=0"
+
+concurrency:
+  # See notes in `pull-request.yaml`
+  group: ${{ github.workflow }}-${{ github.ref }}-check-links
+  cancel-in-progress: true
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: baptiste0928/cargo-install@v2
+        with:
+          crate: mdbook
+
+      # the link checker
+      - uses: baptiste0928/cargo-install@v2
+        with:
+          crate: hyperlink
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: 0.8.1
+          save-if: ${{ github.ref == 'refs/heads/main' }}
+
+      - uses: arduino/setup-task@v1
+        with:
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Build the mdbook
+        run: mdbook build web/book/
+
+      - name: Check links
```

Comment:
One option would be to add this to the existing workflow, or to call the existing workflow as part of this. But no strong view.

Reply:
Though possibly moving the _job_ into `build-web.yaml` would be a reasonable consolidation — it would run slightly too often, but not by much.

Reply:
I am planning on linking relevant pages of the book from doc comments in `prql_ast` in the future ... at which point I want to expand this CI job to check these links as well ... so I think it would be best to keep these separate since they have different path requirements and do largely different things.